rspace/src/test/scala/coop/rchain/rspace/test/InMemoryStore.scala

```diff
@@ -2,185 +2,186 @@ package coop.rchain.rspace.test
 
 import java.nio.charset.StandardCharsets
 
+import cats.implicits._
 import coop.rchain.crypto.hash.Blake2b256
 import coop.rchain.rspace.examples._
 import coop.rchain.rspace.history.{Blake2b256Hash, Trie}
 import coop.rchain.rspace.internal._
 import coop.rchain.rspace.util.dropIndex
 import coop.rchain.rspace.{IStore, ITestableStore, Serialize, StoreCounters, StoreEventsCounter}
 import javax.xml.bind.DatatypeConverter.printHexBinary
+import scodec.Codec
+import scodec.bits.BitVector
+import coop.rchain.shared.AttemptOps._
 
 import scala.collection.immutable.Seq
 import scala.collection.mutable
 
 class InMemoryStore[C, P, A, K] private (
-    _keys: mutable.HashMap[String, Seq[C]],
-    _waitingContinuations: mutable.HashMap[String, Seq[WaitingContinuation[P, K]]],
-    _data: mutable.HashMap[String, Seq[Datum[A]]],
-    _joinMap: mutable.HashMap[C, Seq[Seq[C]]],
+    _dbGNATs: mutable.Map[Blake2b256Hash, GNAT[C, P, A, K]],
+    _dbJoins: mutable.HashMap[C, Seq[Seq[C]]]
 )(implicit sc: Serialize[C], sk: Serialize[K])
     extends IStore[C, P, A, K]
     with ITestableStore[C, P] {
 
-  private[rspace] type H = String
-
-  private[rspace] type T = Unit
+  private implicit val codecC: Codec[C] = sc.toCodec
+  private implicit val codecK: Codec[K] = sk.toCodec
 
   val eventsCounter: StoreEventsCounter = new StoreEventsCounter()
 
-  private[rspace] def hashChannels(cs: Seq[C]): H =
-    printHexBinary(InMemoryStore.hashBytes(cs.flatMap(sc.encode).toArray))
+  private[rspace] type H = Blake2b256Hash
 
-  private[rspace] def putCs(txn: T, channels: Seq[C]): Unit =
-    _keys.update(hashChannels(channels), channels)
+  private[rspace] type T = Unit
 
-  private[rspace] def getChannels(txn: T, s: H) =
-    _keys.getOrElse(s, Seq.empty[C])
+  private[rspace] override def hashChannels(channels: Seq[C]): H =
```

Comment:
that's a topic - the API changes (two new methods, one removed in the past week).
new methods are obvious, but removed ones - no one yells if an out-of-date method is left behind.
this way we get a compiler error.
I don't mind, just thought it would make it easier on us.

will ofc remove.